### ISSUE TITLE
[Bug] Setting IsVisible=true hides views that do not fit on the screen

### ISSUE THREAD

**Issue by IgorKravchenko10** [xamarin/Xamarin.Forms#14398]
### Description
I have created a page with a big count of views. Some of them do not fit on the screen. I also have a Switch to show/hide some of those views. When I toggle Switch on the views that do not fit on the screen will be invisible.
### Steps to Reproduce

1. Create page with several views.
2. Add Switch to show/hide some views.
3. Place some views after views that will be hidden/shown by Switch
4. Toggle switch on to show additional views.
5. Those views will be visible but some views below will dissapear.

Like this:
```
<ScrollView>
            <StackLayout Spacing="3" Padding="15">
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />
                <Entry Placeholder="Default Text" FontSize="Medium" />

                <Switch IsToggled="{Binding IsAdditionalContentVisible}"></Switch>

                <Entry Placeholder="Additional Entry" FontSize="Medium" IsVisible="{Binding IsAdditionalContentVisible}"/>
                <Entry Placeholder="Additional Entry" FontSize="Medium" IsVisible="{Binding IsAdditionalContentVisible}"/>
                <Entry Placeholder="Additional Entry" FontSize="Medium" IsVisible="{Binding IsAdditionalContentVisible}"/>
                <Entry Placeholder="Additional Entry" FontSize="Medium" IsVisible="{Binding IsAdditionalContentVisible}"/>

                <!--Some of these entries will be invisible-->
                <Entry Placeholder="Text" FontSize="Medium" />
                <Entry Placeholder="Text" FontSize="Medium" />
                <Entry Placeholder="Text" FontSize="Medium" />
                <Entry Placeholder="Text" FontSize="Medium" />
                
                <StackLayout Orientation="Horizontal">
                    <Button Text="Cancel" Command="{Binding CancelCommand}" HorizontalOptions="FillAndExpand"></Button>
                    <Button Text="Save" Command="{Binding SaveCommand}" HorizontalOptions="FillAndExpand"></Button>
                </StackLayout>
            </StackLayout>
        </ScrollView>
```

### Expected Behavior
All views must be visible after toggling Switch on.
### Actual Behavior
Those views that are not fit on the screen will be invisible.
### Basic Information

- Version with issue: all
- Last known good version: any
- Platform Target Frameworks: 
  - Android: 11 
- Affected Devices: all

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
Microsoft Visual Studio Enterprise 2019 (2)
Version 16.10.3
VisualStudio.16.Release/16.10.3+31424.327
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Visual C++ 2019   00433-90000-00004-AA789
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.10.526.50910
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   16.10.526.50910
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.10.526.50910
Azure App Service Tools v3.0.0

C# Tools   3.10.0-4.21318.11+7ceb633154acb9d716fd3eb2b6df1a0468d8e416
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

CreateLayoutWizard   1.0
Create layout wizard.

DevExpress Reporting Extension   1.0
A Visual Studio extension that invokes the Report Designer editor for report definition REPX files.

DevExpress.DeploymentTool   1.0
A useful tool for deploying DevExpress assemblies.

DevExpress.Win.LayoutAssistant Extension   1.0
DevExpress.Win.LayoutAssistant Visual Studio Extension Detailed Info

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.10.15 (552afdf)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.10.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Razor (ASP.NET Core)   16.1.0.2122504+13c05c96ea6bdbe550bd88b0bf6cdddf8cde1725
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   16.0.62106.24090
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30429.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.10.0-4.21318.11+7ceb633154acb9d716fd3eb2b6df1a0468d8e416
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.10.0-beta.21262.7+1b23bbeda88ea3cb9be9af777f4c99fa8663df81
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.10.000.234 (d16-10@ecaf29b)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.10.0.115 (remotes/origin/c750fbf1bde3c720d077f51640fe197c6dac7cbe@c750fbf1b)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.10.5 (355b57a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.3.0.4 (d16-10/ae14caf)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: b4a3858
    Java.Interop: xamarin/java.interop/d16-10@f39db25
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.35.4@85460d3
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-10@c5732a0


Xamarin.iOS and Xamarin.Mac SDK   14.20.0.24 (c4b89cddb)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

</details>

### Screenshots

![Screenshot_20210706-210920](https://user-images.githubusercontent.com/8657879/124653351-de91cf80-dea5-11eb-90eb-13a209d0cbd5.jpg)
![Screenshot_20210706-210931](https://user-images.githubusercontent.com/8657879/124653352-de91cf80-dea5-11eb-88e9-f29f1d05cab9.jpg)
![Screenshot_20210706-210936](https://user-images.githubusercontent.com/8657879/124654051-b9519100-dea6-11eb-8d94-e1446bc88652.jpg)

### Reproduction Link

[IsVisibleBugSample.zip](https://github.com/xamarin/Xamarin.Forms/files/6772228/IsVisibleBugSample.zip)

### Workaround

Type some text to entry. Then views will be shown.

**Comment by holecekp** [comment]
For a workaround, see https://github.com/xamarin/Xamarin.Forms/issues/14918#issuecomment-1013697345. The workaround has been proposed originally for another ScrollView bug (https://github.com/xamarin/Xamarin.Forms/issues/13597) but I have tested that it works also for this issue.

**Comment by jfversluis** [comment]
@IgorKravchenko10 if you are still with us, a PR (#15067) for this is open now, would you be able to grab the NuGet [as described here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) and let us know if this fixes this issue? That will greatly speed up the review process.

Besides verifying if this particular issue is fixed also be sure to check other scenarios in the same area to make sure that this fix doesn't accidentally has side-effects 🙂 

Thanks!

**Comment by jfversluis** [comment]
This should be fixed by #15076 and part of SR10
